sample-compose/app/src/main/java/co/nimblehq/sample/compose/ui/screens/home/HomeComposeScreen.kt

```diff
@@ -1,29 +1,64 @@
 package co.nimblehq.sample.compose.ui.screens.home
 
-import androidx.compose.foundation.layout.Box
-import androidx.compose.foundation.layout.fillMaxSize
+import android.widget.Toast
+import androidx.compose.foundation.layout.*
 import androidx.compose.material.CircularProgressIndicator
 import androidx.compose.material.Scaffold
-import androidx.compose.runtime.Composable
+import androidx.compose.runtime.*
 import androidx.compose.ui.Alignment
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
 import androidx.compose.ui.tooling.preview.Preview
+import androidx.hilt.navigation.compose.hiltViewModel
 import co.nimblehq.sample.compose.R
+import co.nimblehq.sample.compose.lib.IsLoading
 import co.nimblehq.sample.compose.model.UiModel
+import co.nimblehq.sample.compose.ui.AppDestination
 import co.nimblehq.sample.compose.ui.screens.AppBar
-import co.nimblehq.sample.compose.ui.theme.ComposeSampleTheme
+import co.nimblehq.sample.compose.ui.theme.ComposeTheme
+import co.nimblehq.sample.compose.ui.userReadableMessage
 
 // TODO: Rename to 'HomeScreen'
 @Composable
 fun HomeComposeScreen(
+    viewModel: HomeComposeViewModel = hiltViewModel(),
+    navigator: (destination: AppDestination) -> Unit
+) {
+    val context = LocalContext.current
+    LaunchedEffect(Unit) {
+        viewModel.error.collect { error ->
+            val message = error.userReadableMessage(context)
+            Toast.makeText(context, message, Toast.LENGTH_SHORT).show()
+        }
+    }
+    LaunchedEffect(viewModel) {
+        viewModel.navigator.collect { destination -> navigator(destination) }
+    }
+
+    val showLoading: IsLoading by viewModel.showLoading.collectAsState()
+    val uiModels: List<UiModel> by viewModel.uiModels.collectAsState()
+
+    HomeScreenContent(
+        uiModels = uiModels,
+        showLoading = showLoading,
+        onItemClick = viewModel::navigateToSecond
+    )
+}
+
+@Composable
+private fun HomeScreenContent(
```

Comment:
Since you created a new `HomeScreenContent` here, I think it makes sense to rename this class to `HomeScreen` altogether. What do you think?

Reply:
@lydiasama renaming this class file name to HomeScreen with over-half file changes will break git history (removed > added new file) and messy git diff in this PR. I renamed it in the next PR https://github.com/nimblehq/android-templates/pull/372/files#diff-190b9b9927a540b81a9f5d2f5014dbb5a57ae16dc60fbdde8d29580fd97a5654.